### ISSUE TITLE
Can't play content with 48k rate

### ISSUE THREAD

**Issue by msmucr** [librespot-org/librespot-java#240]
Hello,

librespot-java release 1.4 throws following exception when playing certain podcast episodes.
I honestly wasn't even aware, there is some Spotify content served with native 48k rate, but it looks like that can also happen.

```
2020-07-02 00:15:10,919 FATAL Log4JUncaughtExceptionHandler:15 - [player-queue-7173442]
java.lang.UnsupportedOperationException: From 'PCM_SIGNED 48000.0 Hz, 16 bit, stereo, 4 bytes/frame, little-endian' to 'PCM_SIGNED 44100.0 Hz, 16 bit, stereo, 4 bytes/frame, little-endian'
        at xyz.gianlu.librespot.player.codecs.StreamConverter.converter(StreamConverter.java:40) ~[librespot-core-jar-with-dependencies-jdk8.jar:1.4.1-SNAPHOST]
        at xyz.gianlu.librespot.player.codecs.Codec.setAudioFormat(Codec.java:95) ~[librespot-core-jar-with-dependencies-jdk8.jar:1.4.1-SNAPHOST]
        at xyz.gianlu.librespot.player.codecs.Mp3Codec.<init>(Mp3Codec.java:31) ~[librespot-core-jar-with-dependencies-jdk8.jar:1.4.1-SNAPHOST]
        at xyz.gianlu.librespot.player.playback.PlayerQueueEntry.load(PlayerQueueEntry.java:112) ~[librespot-core-jar-with-dependencies-jdk8.jar:1.4.1-SNAPHOST]
        at xyz.gianlu.librespot.player.playback.PlayerQueueEntry.run(PlayerQueueEntry.java:248) ~[librespot-core-jar-with-dependencies-jdk8.jar:1.4.1-SNAPHOST]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:1.8.0_251]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:1.8.0_251]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_251]
```
An example of such episode for possible reproduction:
https://open.spotify.com/episode/6dh5gAXM6enMzVbHiPWEBU

The behavior is consistent at Java 8 and Java 11 JREs, at Windows, macOS or Linux. In general audio devices at Windows and MacOS should accept whatever common sample rate and automatically insert system resampler to adjust to actual rate of physical device. I also tried to somehow workaround the issue on Linux with rate conversion plugin in ALSA setup, but it didn't helped.

Thank you!

**Comment by devgianlu** [comment]
The problem here is that librespot forces `PCM_SIGNED 44100.0 Hz, 16 bit, stereo, 4 bytes/frame, little-endian` without trying to open a line for the other format. I'm changing that to hopefully make use of rate conversion plugins.

**Comment by devgianlu** [comment]
@msmucr Should be fixed. Let me know.

**Comment by msmucr** [comment]
@devgianlu Great, thanks a lot for the fix. 48k content there was quite surprise for me.
I've just tried the fix and it works.
At Windows it will play at current working rate of audio device, system level resampler is employed.
The same thing happens on macOS with Core Audio, it also automatically resample according to working rate of the device.
At Linux via ALSA it switches hardware sample rate of audio device.. I successfully tested it as desktop with RME card and on Raspberry Pi with own BCM audio interface. 
If such switching of rate isn't feasible for any reason, it's possible to insert rate parameter for default ALSA slave device, so everything gets resampled from to 44,1k, when desired. I've also tested that with no issues.

Not sure which implications this change would have for someone, who use librespot-java via FIFO pipe output (guessing he would need to also relaunch some other process reading from the pipe with different raw PCM format parameters after possible rate change to 48k), but I haven't tried it so far and direct output to audio interfaces works well.

Thanks again!
